src/dotnet/CarbonAware.WebApi/Controllers/CarbonAwareController.cs

```diff
@@ -1,58 +1,99 @@
-using CarbonAware.Config;
 using CarbonAware.Model;
 using Microsoft.AspNetCore.Mvc;
-using HttpGetAttribute = Microsoft.AspNetCore.Mvc.HttpGetAttribute;
 
 namespace CarbonAware.WebApi.Controllers;
 
 [ApiController]
-[Microsoft.AspNetCore.Mvc.Route("emissions")]
+[Route("emissions")]
 public class CarbonAwareController : ControllerBase
 {
     private readonly ILogger<CarbonAwareController> _logger;
-    private ICarbonAwarePlugin _plugin;
-    private ServiceManager _serviceManager;
-    private IConfigManager _configManager;
+    private readonly ICarbonAware _plugin;
 
-    public CarbonAwareController(ILogger<CarbonAwareController> logger)
+    public CarbonAwareController(ILogger<CarbonAwareController> logger, ICarbonAware plugin)
     {
-        _logger = logger;
-
-        _configManager = new ConfigManager("carbon-aware.json");
-        _serviceManager = new ServiceManager(_configManager);
-        var pluginService = _serviceManager.ServiceProvider.GetService<ICarbonAwarePlugin>();
-
-        if (pluginService is not null)
-        {
-            _plugin = pluginService;
-        }
-        else
-        {
-            throw new Exception("Services are not configured properly.  Could not find plugin service.");
-        }
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _plugin = plugin ?? throw new ArgumentNullException(nameof(plugin));
     }
 
+    [Produces("application/json")]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<EmissionsData>))]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [HttpGet("bylocations/best")]
-    public IEnumerable<EmissionsData> GetBestEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
+    public async Task<IActionResult> GetBestEmissionsDataForLocationsByTime([FromQuery(Name = "locations")] string[] locations, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        var response = _plugin.GetBestEmissionsDataForLocationsByTime(locations.ToList(), time ?? DateTime.Now, toTime, durationMinutes);
-
-        return response;
+        var props = new Dictionary<string, object>() {
+            { CarbonAwareConstants.Locations, locations.ToList() },
+            { CarbonAwareConstants.Start, time ?? DateTime.Now },
+            { CarbonAwareConstants.Duration, durationMinutes },
+            { CarbonAwareConstants.Lowest, true }
```

Comment:
`best` vs `lowest`  I actually prefer `best` because it presumes less about the data source of the plugin.

For example, if your data source has ratings 0 - 100 where 100 is the best (aka lowest) carbon intensity, the semantics of passing `lowest` to the 0-100 rating scale and getting back 99 is confusing.  It obviously is fine when we're talking about raw carbonIntensity measurements, but there are many ways to communicate emissions data.